src/main/java/org/dspace/ctask/replicate/TransmitAIP.java

```diff
@@ -63,17 +67,45 @@ public void init(Curator curator, String taskId) throws IOException {
     @Override
     public int perform(DSpaceObject dso) throws IOException
     {
+        if (skipList != null) {
+            List<String> skipIds = createSkipList(skipList);
+            for (String id : skipIds) {
+                if (id.trim().contentEquals(dso.getHandle())) {
+                    String msg = "This item is in the replicate skiplist: " + dso.getHandle();
+                    setResult(msg);
+                    report(msg);
+                    return Curator.CURATE_SKIP;
+                }
+            }
+        }
+
         ReplicaManager repMan = ReplicaManager.instance();
             
         Packer packer = PackerFactory.instance(dso);
         try
         {
             File archive = packer.pack(repMan.stage(storeGroupName, dso.getHandle()));
-            String msg = "Created AIP: '" + archive.getName() + 
-                         "' size: " + archive.length();
-            repMan.transferObject(storeGroupName, archive);
-            setResult(msg);
-            return Curator.CURATE_SUCCESS;
+            long size = repMan.transferObject(storeGroupName, archive);
+            // The DuraCloud transmission failed. Currently, curate tasks
+            // suspend processing if either CURATE_FAIL or CURATE_ERROR are returned.
+            if (size == -1) {
```

Comment:
Would it be useful for transferObject to throw an exception instead of returning a magic number?  Can it provide information about *how* the transmission failed, to be added to the report?